helpers/test-helper.js

```diff
@@ -408,17 +408,37 @@ export async function submitAndForwardTimeToDispute(clients, test) {
   await forwardTime(MINING_CYCLE_DURATION / 2, test);
 }
 
+export async function runBinarySearch(client1, client2) {
+  // Loop while doing the binary search, checking we were successful at each point
+  // Binary search will error when it is complete.
+  let noError = true;
+  while (noError) {
+    let transactionObject;
+    transactionObject = await client1.respondToBinarySearchForChallenge(); // eslint-disable-line no-await-in-loop
+    let tx = await web3GetTransactionReceipt(transactionObject.hash); // eslint-disable-line no-await-in-loop
+    if (!tx.status) {
+      noError = false;
+    }
+    transactionObject = await client2.respondToBinarySearchForChallenge(); // eslint-disable-line no-await-in-loop
+    tx = await web3GetTransactionReceipt(transactionObject.hash); // eslint-disable-line no-await-in-loop
+    if (!tx.status) {
+      noError = false;
+    }
+  }
+}
+
 export async function getActiveRepCycle(colonyNetwork) {
   const addr = await colonyNetwork.getReputationMiningCycle(true);
   const repCycle = await IReputationMiningCycle.at(addr);
   return repCycle;
 }
 
-export async function advanceMiningCycleNoContest(colonyNetwork, test, miningClient = undefined, minerAddress) {
```

Comment:
Part of that was that the miner/user comes after `test: this` so I figured why not continue with the explicit assignment. For the client names, I figured that `{good,bad}Client` was a good convention, so for tests without good/bad, let's just have them be `client`. It's not amazing justification, I was mostly just thinking "let's make usage consistent according to some convention".

Reply:
Also, I thought that the `miningClient: client` pattern serves as a nice parallel to `minerAddress: MAIN_ACCOUNT`, creating a consistency.

Reply:
Fair enough. I'll just note that 

> Part of that was that the miner/user comes after test: this so I figured why not continue with the explicit assignment.
Part of the attraction for passing in objects, not having parameters, is so that you can have the properties of the object in any order.

Reply:
Works for me :shaka: